dataplane/saiserver/switch.go

```diff
@@ -108,7 +112,7 @@ func newSwitch(mgr *attrmgr.AttrMgr, engine switchDataplaneAPI, s *grpc.Server,
 	saipb.RegisterStpServer(s, sw.stp)
 	saipb.RegisterVirtualRouterServer(s, sw.vr)
 	saipb.RegisterHashServer(s, sw.hash)
-	return sw
+	return sw, nil
```

Comment:
or just return &saiSwitch{...}, nil?

Reply:
i can't because i need to register the sw var before returning it